placeholders_test.go

```diff
@@ -0,0 +1,147 @@
+package lorg
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPlaceholderLevel_ReturnsLevelStringRepresentation(t *testing.T) {
+	test := assert.New(t)
+
+	test.Equal("DEBUG", PlaceholderLevel(LevelDebug, "blah"))
+	test.Equal("FATAL", PlaceholderLevel(LevelFatal, "blah"))
+	test.Equal("FATAL", PlaceholderLevel(LevelFatal, ""))
+}
+
+func TestPlaceholderLine_ReturnsCallerLine(t *testing.T) {
+	test := assert.New(t)
+
+	_, _, line, _ := runtime.Caller(0)
+	test.Equal( // +1
+		strconv.Itoa(line+3),                                  // +2
+		fakePlaceholderStack(PlaceholderLine, LevelDebug, ""), // +3
+	) // +4
+	test.Equal( // +5
+		strconv.Itoa(line+7),                                    // +6
+		fakePlaceholderStack(PlaceholderLine, LevelWarning, ""), // +7
+	)
+}
+
+func TestPlaceholderFile_ReturnsCallerFilenameInShortModeByDefault(
+	t *testing.T,
+) {
+	test := assert.New(t)
+
+	_, file, _, _ := runtime.Caller(0)
+
+	test.Equal(
+		filepath.Base(file),
+		fakePlaceholderStack(PlaceholderFile, LevelInfo, ""),
+	)
+}
+
+func TestPlaceholderFile_ReturnsCallerFilenameInShortModeIfValueIsShort(
+	t *testing.T,
+) {
+	test := assert.New(t)
+
+	_, file, _, _ := runtime.Caller(0)
+
+	test.Equal(
+		filepath.Base(file),
+		fakePlaceholderStack(PlaceholderFile, LevelDebug, "short"),
+	)
+}
+
+func TestPlaceholderFile_ReturnsCallerFullFilenameInLongMode(
+	t *testing.T,
+) {
+	test := assert.New(t)
+
+	_, file, _, _ := runtime.Caller(0)
+
+	test.Equal(
+		file,
+		fakePlaceholderStack(PlaceholderFile, LevelDebug, "long"),
+	)
+	test.Equal(
+		file,
+		fakePlaceholderStack(PlaceholderFile, LevelWarning, "long"),
+	)
+}
+
+func TestPlaceholderTime_ReturnsTimestampIfValueIsTimestamp(t *testing.T) {
+	test := assert.New(t)
+
+	test.Equal(
+		fmt.Sprint(time.Now().Unix()),
+		PlaceholderTime(LevelDebug, "timestamp"),
```

Comment:
Yeah, I think about this, but I decided not to do anything with this, because I can't produce race condition with thousands `go test` repeats (I used `seq 1 10000` for this).

BTW, `go test -race` -> `PASS`